app/src/main/java/com/keylesspalace/tusky/fragment/AccountListFragment.kt

```diff
@@ -307,8 +307,8 @@ class AccountListFragment : Fragment(R.layout.fragment_account_list), AccountAct
 
                 val linkHeader = response.headers()["Link"]
                 onFetchAccountsSuccess(accountList, linkHeader)
-            } catch (throwable: Throwable) {
-                onFetchAccountsFailure(throwable)
+            } catch (exception: IOException) {
+                onFetchAccountsFailure(exception)
```

Comment:
There's an `ifExpected` extension function that checks for IO and HttpException, and rethrows otherwise. Would this be better as?

```
return ifExpected(exception) { onFetchAccountsFailure(exception) }
```

?

Reply:
Don't think so. Same functionality, same line count but no need to know what ifExpected is doing.

Reply:
Can we remove `ifExpected` and replace it with more explicit code like this then?

When reading the code it's a bit of additional friction if two different pieces of code appear to want the same effect but achieve it in two different ways. I'm left wondering whether there's some nuance about the choice that I'm missing.

Reply:
I think @charlag introduced it and in the context where it was used at that time it was saving a few lines